crates/noirc_evaluator/src/brillig/brillig_gen/brillig_block.rs

```diff
@@ -304,7 +318,11 @@ impl<'block> BrilligBlock<'block> {
                         dfg,
                     );
                     let param_id = arguments[0];
-                    self.convert_ssa_array_len(param_id, result_register, dfg);
+                    if let Type::Numeric(_) = dfg.type_of_value(param_id) {
```

Comment:
A comment here stating why the first argument could be a field or an array would help. Also I think we are not actually copying the param to the result register

Reply:
Ah we are not copying it. I have included this logic and a comment